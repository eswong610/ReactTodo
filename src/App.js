// import React, { useState, useEffect } from "react";

// import Todo from './components/Todo';
// import Form from './components/Form';
// import FilterButton from './components/FIlterButton';
// import {nanoid} from 'nanoid';
// import { cleanup } from "@testing-library/react";

// const FILTER_MAP = {
//   All: () => true,
//   Active: task => !task.completed,
//   Completed: task => task.completed
// };

// const FILTER_NAMES = Object.keys(FILTER_MAP);

// function App(props) {
//   // const [tasks, setTasks] = useState(props.tasks);
//   const [tasks, setTasks] = useState([]);
//   const [filter, setFilter] = useState('All');

//   useEffect(()=>{
//     const data = JSON.parse(localStorage.getItem('allTasks'));
//     if (data) {
//       setTasks(data);
//     }
//   }, [])
  
//   useEffect(()=>{
//     localStorage.setItem('allTasks', JSON.stringify(tasks))
//   })
//   // useEffect(()=>{
//   //   localStorage.setItem('allTasks', JSON.stringify(tasks))
//   // })

//   const filterList = FILTER_NAMES.map(name => (
//     <FilterButton
//       key={name}
//       name={name}
//       isPressed={name === filter}
//       setFilter={setFilter}
//     />
//   ));

//   function addTask(name) {
//     const newTask = { id: "todo-" + nanoid(), name: name, completed: false};
//     setTasks([...tasks, newTask]);
    
//   }
    
  

//   function toggleTaskCompleted(id) {
//     const updatedTasks = tasks.map(task => {
//       // if this task has the same ID as the edited task
//       if (id === task.id) {
//         // use object spread to make a new object
//         // whose `completed` prop has been inverted
//         return {...task, completed: !task.completed}
//       }
//       return task;
//     });
//     setTasks(updatedTasks);
//   }

//   function deleteTask(id) {
//     const remainingTasks = tasks.filter(task => id !== task.id);
//     setTasks(remainingTasks);
//   }

//   function editTask(id, newName) {
//     const editedTaskList = tasks.map(task => {
//     // if this task has the same ID as the edited task
//       if (id === task.id) {
//         //
//         return {...task, name: newName}
//       }
//       return task;
//     });
//     setTasks(editedTaskList);
//   }

//   function clearAll () {
//     localStorage.clear();
//     setTasks([]);
//   }

//   // const taskList = tasks.map(task => (
//   //   <Todo
//   //       id={task.id}
//   //       name={task.name}
//   //       completed={task.completed}
//   //       key={task.id}
//   //       toggleTaskCompleted = {toggleTaskCompleted}
//   //       deleteTask= {deleteTask}
//   //       editTask={editTask}
//   //     />
//   //   )
//   // );

// const taskList = tasks.filter(task => FILTER_MAP[filter](task)).map(task => (
//     <Todo
//       id={task.id}
//       name={task.name}
//       completed={task.completed}
//       key={task.id}
//       toggleTaskCompleted={toggleTaskCompleted}
//       deleteTask={deleteTask}
//       editTask={editTask}
//     />
// ));

//   return (
//     <div className="todoapp stack-large">
//       <h1>TodoMatic</h1>
//       <Form addTask={addTask}/>
//       <div className="filters btn-group stack-exception">
//        {filterList}
//        <button onClick={clearAll}>Clear Tasks</button>
       
//       </div>
//       <h2 id="list-heading">
//         {tasks.length} tasks remaining 
//       </h2>
//       <ul
//         role="list"
//         className="todo-list stack-large stack-exception"
//         aria-labelledby="list-heading"
//       >
//        {/* <Todo name="Eat" completed={true} id="todo-0"/>
//        <Todo name="Sleep" completed={false} id="todo-1"/>
//        <Todo name="Repeat" completed={false} id="todo-2"/> */}
//         {taskList}
       
//       </ul>
//     </div>
//   );

// }
// export default App;


import { Route, Switch} from 'react-router-dom';
import './App.css';

import Home from './components/Home'
import Navbar from './components/Navbar'
import Error from './components/Error'
import Api from './components/Api';
import About from './components/About';

import React from 'react'
import ThemeSwitcher from './components/ThemeSwitcher';

function App(props) {
  return (
    <div>
      <ThemeSwitcher />
      <Navbar />
      <Switch>
        <Route path='/' exact component={Home} />
        <Route path='/about' component={About}/>
        <Route path = '/api' component={Api}/>
        <Route component={Error}/>
      </Switch>
    </div>
  )
}

export default App
