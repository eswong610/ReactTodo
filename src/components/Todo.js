import React, {useState} from "react";

export default function Todo(props) {
    const [isEditing, setEditing] = useState(false);
    const [newName, setNewName] = useState("");

    function handleChange (e){
        setNewName(e.target.value);
    }

    function handleSubmit(e) {
        e.preventDefault();
        props.editTask(props.id, newName);
        setNewName("");
        setEditing(false);
      }

    const editingTemplate = (
        <form className="stack-small" onSubmit={handleSubmit}>
          <div className="form-group">
            <label className="todo-label" htmlFor={props.id}>
              New name for {props.name}
            </label>
            <input id={props.id} className="todo-text" type="text" onChange={handleChange} data-testid="edit-input"/>
          </div>
          <div className="btn-group">
            <button type="button" className="btn todo-cancel" onClick={()=>setEditing(false)}>
              Cancel
              <span className="visually-hidden">renaming {props.name}</span>
            </button>
            <button type="submit" className="btn btn__primary todo-edit" >
              Save
              <span className="visually-hidden">new name for {props.name}</span>
            </button>
          </div>
        </form>
      );

    const viewTemplate = (
        <div className="stack-small">
          <div className="c-cb">
              {/* <input
                id={props.id}
                type="checkbox"
                defaultChecked={props.completed}
                onChange={() => props.toggleTaskCompleted(props.id)}
              /> */}
              <div style={{display: 'flex', margin: '1rem'}}>
                  <input type="radio" id={props.id} name={ `status${props.id}` } checked = {props.status=== 0} value="0" onChange={() => props.toggleTaskCompleted(props.id, 0)}/>
                  <label for="male">Not Started</label><br></br>
                  <input type="radio" id={props.id} name={ `status${props.id}` } checked = {props.status === 1}value="1" onChange={() => props.toggleTaskCompleted(props.id, 1)}/>
                  <label for="female">Started</label>
                  <input type="radio" id={props.id} name={ `status${props.id}` } checked ={props.status === 2} value="2" onChange={() => props.toggleTaskCompleted(props.id, 2)}/>
                  <label for="other">Completed</label>
              </div>

              <label className="todo-label" htmlFor={props.id} >
                {props.name}
              </label>
            </div>
            <div className="btn-group">
              <button type="button" className="btn" onClick={() => setEditing(true)} data-testid="edit-button">
                Edit <span className="visually-hidden">{props.name}</span>
              </button>
              <button
                type="button"
                className="btn btn__danger"
                onClick={() => props.deleteTask(props.id)}
              >
                Delete <span className="visually-hidden">{props.name}</span>
              </button>
            </div>
        </div>
      );
    
      // console.log(isEditing + 'is editing');
      
      return(
        <li data-testid="isEditing">{isEditing ? editingTemplate : viewTemplate}</li>
      )
    
  }