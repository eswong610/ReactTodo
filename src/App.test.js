import React, { createElement } from 'react';
// import { ReactDOM, domrender }from 'react-dom';
import renderer from 'react-test-renderer';
import { render, cleanup, fireEvent } from '@testing-library/react';
import  "@testing-library/jest-dom/extend-expect";
import App from './App';
import Todo from './components/Todo';
import Form from './components/Form';
import FilterButton from './components/FIlterButton';

// test('renders learn react link', () => {
//   const { getByText } = render(<App />);
//   const linkElement = getByText(/learn react/i);
//   expect(linkElement).toBeInTheDocument();
// });

// test('renders without crashing', ()=>{
//     const div = createElement('div');
//     ReactDOM.render(<App />, div)
// })

afterEach(cleanup);
  
  describe('the todo list', () => {
  
    it('should be a button that says edit', () => {
        const { getByTestId } = render(<Todo />); 
        expect(getByTestId('edit-button')).toHaveTextContent(/Edit/i);  
       });


    it('has edit input once edit button is clicked', () => {
        const { queryByTestId, getByTestId } = render(<Todo />); 
        fireEvent.click(getByTestId('edit-button'))
        expect(queryByTestId(/edit-input/i)).toBeTruthy();
    });

    it("form has proper label", () => {
        // const sampleTask = (name) => {
        //     return name;
        // }
        // const { getByTestId } = render(<Form addTask={sampleTask}/>)
        // fireEvent.change(getByTestId('form-input'),  { target: { value: 'new' } } );
        // fireEvent.click(getByTestId("form-submit"))
        // expect(getByTestId('form-input')).toHaveTextContent("new")
        const {getByTestId} = render(<Form/>)
        expect(getByTestId('form-label')).toHaveTextContent('What needs to be done?')
    })

    it("app snapshot", () => {
        const component = renderer.create(<FilterButton />)
        const tree = component.toJSON;
        expect(tree).toMatchSnapshot();
    })
  });
