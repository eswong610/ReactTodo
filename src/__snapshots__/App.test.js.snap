// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`the todo list app snapshot 1`] = `[Function]`;
